src/AWS.Deploy.Orchestration/Orchestrator.cs

```diff
@@ -131,16 +134,22 @@ public async Task DeployRecommendation(CloudApplication cloudApplication, Recomm
             _interactiveService.LogMessageLine(string.Empty);
             _interactiveService.LogMessageLine($"Initiating deployment: {recommendation.Name}");
 
-            if (recommendation.Recipe.DeploymentType == DeploymentTypes.CdkProject)
-            {
-                _interactiveService.LogMessageLine("AWS CDK is being configured.");
-                await _cdkManager.EnsureCompatibleCDKExists(Constants.CDK.DeployToolWorkspaceDirectoryRoot, Constants.CDK.MinimumCDKVersion);
-            }
-
             switch (recommendation.Recipe.DeploymentType)
             {
                 case DeploymentTypes.CdkProject:
-                    await _cdkProjectHandler.CreateCdkDeployment(_session, cloudApplication, recommendation);
+                    var cdkProject = await _cdkProjectHandler.ConfigureCdkProject(_session, cloudApplication, recommendation);
+                    _interactiveService.LogMessageLine("AWS CDK is being configured.");
+
+                    if (_cdkVersionDetector == null)
+                    {
+                        throw new InvalidOperationException($"{nameof(_cdkVersionDetector)} must not be null.");
+                    }
```

Comment:
can we do this check earlier and error out before we configure the cdk project.

Reply:
nope that's not going to work for following reasons, we need csproj file in order to detect which is only available once the deployment is generated. It could be in temp dir or customer codebase.

Reply:
I'm not sure i get what you mean. _cdkVersionDetector will be null on line 140 and 143. So instead of checking it on 143, I'd like to check it on before 140

Reply:
I don't like that either, why even dependencies are marked as nullable which is the root cause of the problem here. Moving the code two lines above does not solve the problem.

Reply:
Moved to as first line to unblock this PR. We have backlog now DOTNET-5358.